### PR TITLE
adding random uniform initialization

### PR DESCRIPTION
**Pull request by FollowKenny**
This PR solves #398. It adds :

- random_uniform method to all backends. This provides a way to initiate a tensornetwork with values sampled from a uniform distribution.
- tests for random_uniform for all backends

**Caution** : I added a test to validate the behaviour of the random uniform initialization for all backends but tensorflow. Indeed the seed system in tensorflow (graph/op seed) messed with my brain and I could not consistently init a tensornetwork directly as tf object and via the backend method.
As this test is a bit of an overkill, I'm still pushing as is. If you guys have an idea...

### CODE REVIEW

**Comment by FollowKenny** [comment]
@mganahl Ready for review I believe.

**Comment by Thenerdstation** [comment]
That's ok, the TF random module I believe wasn't designed to be deterministic, so not much you can do.

File: tensornetwork/backends/base_backend.py [hunk 1]
```diff
@@ -265,6 +265,25 @@ def randn(self,
     raise NotImplementedError("Backend '{}' has not implemented randn.".format(
         self.name))
 
+  def random_uniform(self,
+                     shape: Tuple[int, ...],
+                     boundaries: Optional[Tuple[float, float]] = (0.0, 1.0),
+                     dtype: Optional[Type[np.number]] = None,
+                     seed: Optional[int] = None) -> Tensor:
+    """Return a random-uniform-matrix of dimension `dim`
```
Comment (Thenerdstation) [inline tensornetwork/backends/base_backend.py]: Drop the dashes.

Comment (Thenerdstation) [reply tensornetwork/backends/base_backend.py]: nit: Add period

File: tensornetwork/backends/base_backend.py [hunk 2]
```diff
@@ -265,6 +265,25 @@ def randn(self,
...
+       Depending on specific backends, `dim` has to be either an int 
+       (numpy, torch, tensorflow) or a `ShapeType` object 
+       (for block-sparse backends). Block-sparse
+       behavior is currently not supported
+       Args:
+         shape (int): The dimension of the returned matrix.
+         boundaries (tuple) : the boundaries of the uniform distribution
+         dtype: The dtype of the returned matrix.
+         seed:  The seed for the random number generator
```
Comment (Thenerdstation) [inline tensornetwork/backends/base_backend.py]: Be sure to set a Returns.

**Comment by FollowKenny** [comment]
Nit picks done, should be good to go.

**Pull request merged**